apps/nextjs-website/src/lib/api.ts

```diff
@@ -46,3 +47,24 @@ export const getFooterProps = async (): Promise<FooterProps> => {
   const footer = await getFooter(appEnv);
   return makeFooterProps(footer);
 };
+
+// Return PageProps for a specific page ID
+export const getPageProps = async (
+  path: string
+): Promise<PageData['data']['attributes'] | null> => {
+  const allPages = await getAllPages();
+
+  const pageID = allPages.filter((page) => path === page.slug.join('/'))[0]?.id;
+
+  if (!pageID) {
+    return null;
+  }
+
+  const {
+    data: { attributes },
+  } = await getPage(pageID, {
+    config: appEnv.config,
+    fetchFun: appEnv.fetchFun,
```

Comment:
The documentation linked is for the Navigation plugin.

`getPage` fetches from [Strapi's default REST API](https://docs.strapi.io/dev-docs/api/rest), which only accepts ID.

Reply:
After some testing, I managed to use the Navigation Plugin to fetch the page's data using its path, the solution has a couple of problems though.

1. Fetching through the Navigation Plugin, we are assuming that the page's slug is equal to the Navigation Item's path, which is not always going to be true and will differ as soon as a user changes a page's slug (which is not unrealistic).
2. The plugin doesn't seem to support a list of parameters like `?populate=ctaButtons,image,link`. This forces us to explicitly list every property we want to render, resulting in a less-than-pretty URL such as this one `http://localhost:1337/api/navigation/render/1?path=/home&populate[sections][populate][0]=ctaButtons&populate[sections][populate][1]=image&populate[sections][populate][2]=background&populate[sections][populate][3]=items&populate[sections][populate][4]=link&populate[sections][populate][5]=steps`

For these reasons I would not opt for this solution.

Reply:
However, if we wish to reduce the number of calls to the server we could fetch all pages with a call populating all related fields of all pages: `http://localhost:1337/api/navigation/render/1?type=FLAT&populate[sections][populate][0]=ctaButtons&populate[sections][populate][1]=image&populate[sections][populate][2]=background&populate[sections][populate][3]=items&populate[sections][populate][4]=link&populate[sections][populate][5]=steps`

We already know that the result of this call will get cached internally by NextJS (as it already happens in the call to `getAllPages`) so we should be able to avoid the following `getPage` call entirely by simply extracting the value from the `related` field that's already returned by `getAllPages`.

Let me know if this solution seems desirable and I will implement it asap.